subscription/src/main/java/org/killbill/billing/subscription/api/svcs/DefaultSubscriptionInternalApi.java

```diff
@@ -136,70 +136,27 @@ public DefaultSubscriptionInternalApi(final SubscriptionDao dao,
-                                               final PlanPhaseSpecifier spec,
-                                               final List<PlanPhasePriceOverride> overrides,
-                                               final DateTime requestedDateWithMs,
-                                               final boolean isMigrated,
-                                               final InternalCallContext context) throws SubscriptionBaseApiException {
-        try {
-            if (bundle == null) {
-                throw new SubscriptionBaseApiException(ErrorCode.SUB_CREATE_NO_BUNDLE, null);
-            }
-
-            final DateTime now = clock.getUTCNow();
-            final DateTime effectiveDate = (requestedDateWithMs != null) ? DefaultClock.truncateMs(requestedDateWithMs) : now;
-            /*
-            if (requestedDate.isAfter(now)) {
-                throw new SubscriptionBaseApiException(ErrorCode.SUB_INVALID_REQUESTED_DATE, now.toString(), requestedDate.toString());
-            }
-            */
-
-            final CallContext callContext = internalCallContextFactory.createCallContext(context);
-            final Catalog catalog = catalogInternalApi.getFullCatalog(true, true, context);
-            final PlanPhasePriceOverridesWithCallContext overridesWithContext = new DefaultPlanPhasePriceOverridesWithCallContext(overrides, callContext);
-
-            final Plan plan = catalog.createOrFindPlan(spec, overridesWithContext, effectiveDate);
-            final PlanPhase phase = plan.getAllPhases()[0];
-            if (phase == null) {
-                throw new SubscriptionBaseError(String.format("No initial PlanPhase for Product %s, term %s and set %s does not exist in the catalog",
-                                                              spec.getProductName(), spec.getBillingPeriod().toString(), plan.getPriceListName()));
-            }
-
-            // verify the number of subscriptions (of the same kind) allowed per bundle
-            if (ProductCategory.ADD_ON.toString().equalsIgnoreCase(plan.getProduct().getCategory().toString())) {
-                if (plan.getPlansAllowedInBundle() != -1
-                    && plan.getPlansAllowedInBundle() > 0
-                    && addonUtils.countExistingAddOnsWithSamePlanName(getSubscriptionsForBundle(bundle.getId(), null, context), plan.getName())
-                       >= plan.getPlansAllowedInBundle()) {
-                    // a new ADD_ON subscription of the same plan can't be added because it has reached its limit by bundle
-                    throw new SubscriptionBaseApiException(ErrorCode.SUB_CREATE_AO_MAX_PLAN_ALLOWED_BY_BUNDLE, plan.getName());
-                }
+    private List<SubscriptionSpecifier> verifyAndBuildSubscriptionSpecifiers(final SubscriptionBaseBundle bundle,
+                                                                             @Nullable final EntitlementSpecifier baseOrStandalonePlanSpecifier,
```

Comment:
Let's talk, I'm not sure I follow.

Note that `baseOrStandalonePlanSpecifier` (now renamed `baseOrFirstStandalonePlanSpecifier`) here is only used to compute the bundle start date.

Reply:
What i am trying to say is that all `STANDALONE` subscriptions are treated the same, none of them are special, so from an abstraction point of view having a `firstStandalonePlanSpecifier` seems broken. Then if the goal is the get the bundle startDate, then 2 thoughts come to mind:

1. Note that `BUNDLE` alignment for `STANDALONE` does not make sense -- i suppose this date is for this use case.
2. How do we determine which one is the first to come up with the date, and what is this date used for -- my guess is nothing, because of `1.` ?

Reply:
Tentative fix in 2370dd3e0f08f6c577b7e7e367c2f9377f683f4e.